docs/docs/monitoring/analytics/realtime-markers.mdx

```diff
@@ -0,0 +1,87 @@
+---
+id: realtime-markers
+sidebar_label: Real-time Markers
+title: Real-time Analysis of Markers
+description:
+abstract: Process Markers real-time from Rasa
+    Assistants to track metrics like solution and abandonment rates,
+    add metatags to conversations,
+    or filter conversations for Conversation Driven Development
+---
+
+import useBaseUrl from "@docusaurus/useBaseUrl";
+import RasaProLabel from "@theme/RasaProLabel";
+import RasaProBanner from "@theme/RasaProBanner";
+import { Grid } from "@theme/Grid";
+
+<RasaProLabel />
+
+:::info Available in Rasa Pro from 3.6.0
+:::
+
+<RasaProBanner />
+
+In Rasa Pro, [Markers](../../markers.mdx) is a powerful feature that allows
+you to track and extract custom conversational events. With the Analytics
+Data Pipeline, you can now process these Markers in real-time, enabling
+you to gain valuable insights and enhance the performance of your Rasa
+Assistant. In this guide, we'll explore how to leverage real-time
+analysis of Markers to track solution rate and abandonment in your conversations.
+
+## Defining Markers
+
+Please consult the [Markers](../../markers.mdx/#defining-markers) section of Rasa documentation
+for details about defining Markers.
+
+## Enable Real-time Processing
+
+To enable real-time analysis of Markers in your Rasa project, you need to
+upload the markers to Analytics Data Pipeline with the `rasa markers upload`
+command. This command uploads the marker configuration YAML and starts the
+processing of Markers in the Analytics Data Pipeline. To get started,
+make sure you have the latest version of Rasa installed. Then, open your command line
+interface and navigate to your Rasa project directory. Run the following command:
+
+```
+rasa markers upload --config=<path-to-config-file> -d=<path-to-domain-file> -rasa-pro-services-url=<url>
+```
+
+By default, this command validates the marker configuration file against
+your bot's domain.yml file. To specify a different domain file, use the
+optional `-d` argument.
+
+This command should be run whenever there is a change in the marker
```

Comment:
How do you stop forwarding real-time markers?

Reply:
I'm sorry I didn't understand your question. Did you mean to ask how does one stop uploading the marker-configurations to rasa-pro-services? The marker-configurations are only forwarded to rasa-pro-services (referred as Analytics Data Pipeline in the documentation) when this command is run.

Reply:
I have the impression that this command starts "real time processing" of markers and that it kicks off real-time forwarding of markers to the analytics pipeline by Rasa.

I must be confused about what happens when you run this command.

Does the command do a one time upload of markers to the analytics and then exit?  Does it run and forward markers until you manually stop the command?

Reply:
This command uploads the marker-configuration(aka patterns) files and then exits. Rasa Analytics stores these patterns in a database table called `rasa_patterns`. To remove any pattern, you can remove them from the YAML file and run `rasa markers upload` again.

Rasa Analytics will then process all events from Kafka and extract any markers that it founds in the stream of events. The markers are extracted based on the patterns in `rasa_patterns` table. Extracted markers are stored in `rasa_markers` table